src/serializer/ResidualHeapSerializer.js

```diff
@@ -551,14 +551,61 @@ export class ResidualHeapSerializer {
     }
   }
 
-  _serializeValueArray(val: ObjectValue): BabelNodeExpression {
-    let realm = this.realm;
+  _serializeArrayIndexProperties(array: ObjectValue, remainingProperties: Map<string, PropertyBinding>) {
     let elems = [];
+    // An array's length property cannot be redefined, so this won't run user code
+    const allPropertyLength = remainingProperties.size;
+    for (let i = 0; i < allPropertyLength; i++) {
+      let key = i + "";
+      let propertyBinding = remainingProperties.get(key);
+      if (propertyBinding === undefined) {
+        // Array index should be sequencial.
```

Comment:
I am confused if the array length will always be set correctly. Let's talk in person...